DeprecationWarning: pkg_resources is deprecated as an API

### What happened?

Running in CI with warnings enabled, eth_utils import fails the tests because it uses pkg_resources.

### Code that produced the error

```python
from eth_utils import decode_hex
```


### Full error output

```shell
tests/test_query.py:3: in <module>
    from tronpy import AsyncTron, Tron
tronpy/__init__.py:1: in <module>
    from tronpy.async_contract import AsyncContract
tronpy/async_contract.py:5: in <module>
    from tronpy.contract import Contract, ContractFunctions, ContractMethod, ShieldedTRC20
tronpy/contract.py:5: in <module>
    from eth_utils import decode_hex
../venv/lib/python3.8/site-packages/eth_utils/__init__.py:1: in <module>
    import pkg_resources
../venv/lib/python3.8/site-packages/pkg_resources/__init__.py:121: in <module>
    warnings.warn("pkg_resources is deprecated as an API", DeprecationWarning)
E   DeprecationWarning: pkg_resources is deprecated as an API
```


### Fill this section in if you know how this could or should be fixed

https://setuptools.pypa.io/en/latest/pkg_resources.html
importlib.resources and similar should be used instead

### eth-utils Version

2.1.1

### Python Version

3.8

### Operating System

linux

### Output from `pip freeze`

_No response_

--- a/.gitignore
+++ b/.gitignore
@@ -20,6 +20,7 @@ lib
 lib64
 pip-wheel-metadata
 venv*
+.venv*
 
 # Installer logs
 pip-log.txt

--- a/eth_utils/__init__.py
+++ b/eth_utils/__init__.py
@@ -1,4 +1,12 @@
-import pkg_resources
+try:
+    from importlib.metadata import version as __version
+except ImportError:
+    # Python 3.7
+    def __version(package_name: str) -> str:  # type: ignore
+        from pkg_resources import get_distribution
+
+        return get_distribution(package_name).version
+
 
 from .abi import (  # noqa: F401
     event_abi_to_log_topic,
@@ -97,4 +105,4 @@ from .types import (  # noqa: F401
     is_tuple,
 )
 
-__version__ = pkg_resources.get_distribution("eth-utils").version
+__version__ = __version("eth-utils")

--- a/newsfragments/242.misc.rst
+++ b/newsfragments/242.misc.rst
@@ -1,0 +1,1 @@
+Use ``importlib`` to replace ``pkg_resources`` (deprecated) in Python >= 3.8.

